docs/AsyncWrap/example-trace/trace.js

```diff
@@ -2,37 +2,45 @@
 
 const asyncWrap = process.binding('async_wrap');
 
-asyncWrap.setupHooks(init, before, after);
+asyncWrap.setupHooks(init, before, after, destroy);
 asyncWrap.enable();
 
-// global state variable, that contains the current stack trace
-let currentStack = '';
+// global state variable, that contains the stack traces and the current uid
+const stack = new Map();
+      stack.set(-1, '');
+let currentUid = -1;
 
-function init(provider, uid, parent) {
+function init(uid, provider, parentUid, parentHandle) {
   // When a handle is created, collect the stack trace such that we later
   // can see what involved the handle constructor.
   const localStack = (new Error()).stack.split('\n').slice(1).join('\n');
 
-  // Compute the full stack and store it as a property on the handle object,
-  // such that it can be fetched later.
-  const extraStack = parent ? parent._full_init_stack : currentStack;
-  this._full_init_stack = localStack + '\n' + extraStack;
+  // Compute the full stack and store on the `Map` using the `uid` as key.
+  const extraStack = stack.get(parentUid || currentUid);
+  stack.set(uid, localStack + '\n' + extraStack);
 }
-function before() {
-  // A callback is about to be called, update the `currentStack` such that
+function before(uid) {
+  // A callback is about to be called, update the `currentUid` such that
   // it is correct for when another handle is initialized or `getStack` is
   // called.
-  currentStack = this._full_init_stack;
+  currentUid = uid;
 }
-function after() {
+function after(uid) {
   // At the time of writing there are some odd cases where there is no handle
   // context, this line prevents that from resulting in wrong stack trace. But
   // the stack trace will be shorter compared to what ideally should happen.
-  currentStack = '';
+  currentUid = -1;
+}
+
+function destroy(uid) {
+  // Once the handle is destroy no other handle objects can be created with
```

Comment:
This is the name that was chosen: https://github.com/nodejs/node/blob/master/src/async-wrap.cc#L135